scripts/train.py

```diff
@@ -124,69 +130,89 @@ def main(cfg: TrainConfig) -> None:
     olmo_model = OLMo(cfg.model)
     log.info(f"Total number of parameters: {olmo_model.num_params():,d}")
     log.info(f"Number of non-embedding parameters: {olmo_model.num_params(include_embedding=False):,d}")
-    log.info(f"Peak GPU Memory (MB) before FSDP: {int(peak_gpu_memory() or 0)}")
+    log.info(f"Peak GPU Memory (MB) before {cfg.distributed_strategy}: {int(peak_gpu_memory() or 0)}")
 
     olmo_model.set_activation_checkpointing(cfg.activation_checkpointing)
 
-    # Wrap the model in FSDP.
-    log.info("Wrapping model with FSDP...")
-    wrap_policy = olmo_model.get_fsdp_wrap_policy(cfg.fsdp.wrapping_strategy)
+    if cfg.distributed_strategy == DistributedStrategy.ddp:
+        log.info("Wrapping model with DDP...")
+        assert (
+            cfg.model.init_device == "cuda"
+        ), "DDP does not work with init_device set to anything other than `cuda`."
 
-    if version.parse(torch.__version__) >= version.parse("2.1.0"):
-        # This prevents any parameters from being initialized twice
-        def dummy_init_fn(module: torch.nn.Module) -> None:
-            module.to_empty(device=get_default_device())
+        if cfg.ddp.find_unused_params is True:
+            assert (
+                cfg.ddp.grad_sync_mode == DDPGradSyncMode.micro_batch
+            ), "`find_unused_params` is set to True. DDP needs to synchronize gradients for every micro-batch to avoid errors. Setting `grad_sync_mode` to `micro_batch`."
```

Comment:
Can you throw `OLMoConfigurationError` instead of `AssertionError`? That's our convention.

Reply:
should I throw a configuration error or quietly correct this?

```
if cfg.ddp.find_unused_params is True and cfg.ddp.grad_sync_mode != DDPGradSyncMode.micro_batch:
            cfg.ddp.grad_sync_mode = DDPGradSyncMode.micro_batch
            log.warning("`find_unused_params` is set to True. DDP needs to synchronize gradients for every micro-batch to avoid errors. Setting `grad_sync_mode` to `micro_batch`.") 
```

Reply:
ah we should probably raise an error so that we maintain uniformity and information in the config files.